### ISSUE TITLE
[Feature Request]Move's comments should supported other language's characters

### ISSUE THREAD

**Issue by jolestar** [diem/diem#1074]
# 🚀 Feature Request

## Motivation

Move's comments only allow ASCII  visual characters (0x20 <= c <= 0x7E), user can not use other languages to write comments, such as Chinese.  Comments should support all-natural languages.  

```
// We restrict strings to only ascii visual characters (0x20 <= c <= 0x7E) or a permitted newline
// character--\n--or a tab--\t.
fn strip_comments_and_verify(string: &str) -> Result<String> {
    verify_string(string)?;
    Ok(strip_comments(string))
}
```

**Is your feature request related to a problem? Please describe.**

The functional_tests call strip_comments directly instead of strip_comments_and_verify, so the user can use any characters in test case comments, but the Move script file can not be compiled.

Hope test case and compiler have the same behavior.

## Pitch

**Describe the solution you'd like**

1. Expand the character limit in Move comments.
2. functional_tests use the same method to verify comments.

**Describe alternatives you've considered**

**Are you willing to open a pull request?** (See [CONTRIBUTING](../../CONTRIBUTING.md))
Yes

**Comment by tnowacki** [comment]
It's something we will add for the source language, but it will not be a part of the Move IR. 

The Move IR is not designed for a good developer experience, and it serves as a thin syntactic wrapper over the Move bytecode (with the purpose of testing the bytecode verifier). 

So we won't do (1), but we will however look into (2)

**Comment by jolestar** [comment]
Now, The source language should support no ASCII character.

**Comment by bob-wilson** [comment]
I suppose we should revisit this topic soon.

**Comment by modocache** [comment]
Thanks for re-opening the discussion on this, @jolestar! I think it's important Move support other languages in comments. I sent up #9138 to take a closer look at this issue.

As we discussed there, I think the challenge we face in supporting non-ASCII characters in comments is the difficulty in visually determining what parts of the source code are comments, and what parts are not. I think a good path forward on supporting Unicode comments is to first provide people with Move syntax highlighting tools that can make it clear what is and is not a comment. I'm working on such a tool now, and I'll post updates on next steps here once the tool is ready.
